components/tel/Core/Src/canload.c

```diff
@@ -0,0 +1,72 @@
+#include "canbusload.h"
+
+uint32_t can_total_bits = 0;
+static uint64_t counter_window = 0;
+static uint64_t current_bus_load = 0;
+uint32_t currentIdx = 0;
+uint32_t circularBuffer[CANLOAD_WINDOW_SIZE] = {0};
+
+uint32_t CANLOAD_calculate_message_bits(CAN_msg_t *msg)
```

Comment:
Instead of standardizing the input of the function to be a struct you made, lets pass in just the header of the CAN message which has a type **already defined in the STM32 HAL**.

Reply:
Also as of current `CAN_msg_t` is only something that exists in the old TEL code so currently you would need to create that typedef in this code.